languages/rust/exercises/concept/structs/src/lib.rs

```diff
@@ -0,0 +1,36 @@
+#[derive(Clone, Debug)]
+pub struct User {
+    name: String,
+    age: u32,
+    weight: f32,
+}
+
+impl User {
+    pub fn new(name: String, age: u32, weight: f32) -> Self {
+        unimplemented!()
+    }
+
+    pub fn get_name(&self) -> &str {
+        unimplemented!()
+    }
+
+    pub fn get_age(&self) -> u32 {
+        unimplemented!()
+    }
+
+    pub fn get_weight(&self) -> f32 {
+        unimplemented!()
+    }
+
+    pub fn update_name(&mut self, new_name: String) {
+        unimplemented!()
+    }
+
```

Comment:
Names don't change very often in the real world; for the purposes of this model, we might want to just omit this method to demonstrate why you might use a getter instead of a public field: so that it can't be changed by downstream users.

```suggestion
```